test/cql-pytest/test_filtering.py

```diff
@@ -125,3 +125,16 @@ def test_filter_like_on_desc_column(cql, test_keyspace):
         cql.execute(f"INSERT INTO {table} (a, b) VALUES (1, 'one')")
         res = cql.execute(f"SELECT b FROM {table} WHERE b LIKE '%%%' ALLOW FILTERING")
         assert res.one().b == "one"
+
+# Test that the fact that a column is indexed does not cause us to fetch
+# incorrect results from a filtering query (issue #10300).
+def test_index_with_in_relation(cql, test_keyspace):
+    schema = 'p int, c int, v boolean, primary key (p,c)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        for p, c, v in [(0,0,True),(0,1,False),(0,2,True),(0,3,False),
+                (1,0,True),(1,1,False),(1,2,True),(1,3,False),
+                (2,0,True),(2,1,False),(2,2,True),(2,3,False)]:
+            cql.execute(f"insert into {table} (p,c,v) values ({p}, {c}, {v})")
+        cql.execute(f"create index on {table}(v)")
```

Comment:
I would move the CREATE INDEX statement *before* inserting the data. Otherwise, the index is only built later, asynchronously, and it's not clear what you're testing below.
BUT, if you did see a failure of this test before your fix, I guess it does test the right thing so I can commit it like this.

Reply:
No, you're right, it should be created first. I did see a failure before the fix, but that's either a race or an implementation detail, and I shouldn't rely on either of these.